providers/gitlab/resources/gitlab.go

```diff
@@ -121,3 +121,30 @@ func initGitlabProject(runtime *plugin.Runtime, args map[string]*llx.RawData) (m
 	args = getGitlabProjectArgs(project)
 	return args, nil, nil
 }
+
+// New function to fetch project approval settings
+func (p *mqlGitlabProject) approvalSettings() ([]interface{}, error) {
+	conn := p.MqlRuntime.Connection.(*connection.GitLabConnection)
+
+	projectID := int(p.Id.Data)
+	approvalConfig, _, err := conn.Client().Projects.GetApprovalConfiguration(projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	approvalSettings := map[string]*llx.RawData{
+		"approvalsBeforeMerge":                      llx.IntData(int64(approvalConfig.ApprovalsBeforeMerge)),
+		"resetApprovalsOnPush":                      llx.BoolData(approvalConfig.ResetApprovalsOnPush),
+		"disableOverridingApproversPerMergeRequest": llx.BoolData(approvalConfig.DisableOverridingApproversPerMergeRequest),
+		"mergeRequestsAuthorApproval":               llx.BoolData(approvalConfig.MergeRequestsAuthorApproval),
+		"mergeRequestsDisableCommittersApproval":    llx.BoolData(approvalConfig.MergeRequestsDisableCommittersApproval),
+		"requirePasswordToApprove":                  llx.BoolData(approvalConfig.RequirePasswordToApprove),
+	}
+
+	mqlApprovalSettings, err := CreateResource(p.MqlRuntime, "gitlab.project.approvalSettings", approvalSettings)
+	if err != nil {
+		return nil, err
+	}
+
+	return []interface{}{mqlApprovalSettings}, nil
```

Comment:
it looks like maybe approval settings should be singular? as in `approvalSettings() []gitlab.project.approvalSettings` could be `approvalSettings() gitlab.project.approvalSettings`, since there's only ever one?

if that's true, I think you can change this return to `mqlApprovalSettings.(*mqlGitlabProjectApprovalSettings)` and then set the return of the function to `*mqlGitlabProjectApprovalSettings` after changing `[]gitlab.project.approvalSettings` to `gitlab.project.approvalSettings`

Reply:
Thanks, that's true, there is only one approval settings for the whole project. Changed it as requested.

Reply:
nice, thank you!